### PR TITLE
Change supported crystal version to >= 0.35.0

### PR DESCRIPTION
**Pull request by Sija**
Supersedes #173

### CODE REVIEW

**Review by veelenga — APPROVED** [review]
I went through the release notes and I think it is still compatible with 0.29. But since this compatibility is not that important and there is no much value in supporting it, i'm ok with this change.

File: shard.yml [hunk 1]
```diff
@@ -1,8 +1,6 @@
 name: ameba
 version: 0.13.2
```
Comment (drhuffman12) [inline shard.yml:2]: Should the version be bumped up?

Comment (Sija) [reply shard.yml:2]: Not in this PR, no.

Comment (veelenga) [reply shard.yml:2]: @drhuffman12 do you need a new tagged release? I can issue one.

Comment (Blacksmoke16) [reply shard.yml:2]: @veelenga Yes please.  https://github.com/athena-framework/validator/pull/9/checks?check_run_id=1496168419

Comment (veelenga) [reply shard.yml:2]: @Blacksmoke16 @drhuffman12 published https://github.com/crystal-ameba/ameba/releases/tag/v0.13.3

**Pull request merged**